src/views/DrefApplicationForm/schema.ts

```diff
@@ -280,92 +226,318 @@ const schema: DrefFormSchema = {
+            ifrc_project_manager_name: {},
+            ifrc_project_manager_email: { validations: [emailCondition] },
+            ifrc_project_manager_title: {},
+            ifrc_project_manager_phone_number: {},
+
+            national_society_contact_name: {},
+            national_society_contact_title: {},
+            national_society_contact_email: { validations: [emailCondition] },
+            national_society_contact_phone_number: {},
+            ifrc_emergency_name: {},
+            ifrc_emergency_title: {},
+            ifrc_emergency_email: { validations: [emailCondition] },
+            ifrc_emergency_phone_number: {},
+            media_contact_name: {},
+            media_contact_title: {},
+            media_contact_email: { validations: [emailCondition] },
+            media_contact_phone_number: {},
+            end_date: {},
+            publishing_date: {},
+            glide_code: {},
+
+            // government_requested_assistance_date: {}, // NOTE: Not found in the UI
+            // community_involved: {}, // NOTE: Not found in the UI
         };
 
-        const assessmentAffectedFields = [
-            'did_it_affect_same_area',
-            'did_it_affect_same_population',
-            'did_ns_respond',
-            'did_ns_request_fund',
-            'ns_request_text',
-            'lessons_learned',
-            'dref_recurrent_text',
-            'identified_gaps',
-            'women',
-            'men',
-            'girls',
-            'boys',
-            'event_scope',
-            'assessment_report',
-            'logistic_capacity_of_ns',
-            'pmer',
-            'communication',
-            'needs_identified',
-        ] as const;
-
-        type ConditionalReturnType = Pick<
-            DrefFormSchemaFields,
-            (typeof assessmentAffectedFields)[number]
-        >;
-
-        // TODO: verify schema, add condition for loan type
+        // OVERVIEW
+
+        formFields = addCondition(
+            formFields,
+            formValue,
+            ['disaster_type'],
+            ['is_man_made_event'],
+            (val) => {
+                if (
+                    val?.disaster_type === DISASTER_FIRE
+                    || val?.disaster_type === DISASTER_FLOOD
+                    || val?.disaster_type === DISASTER_FLASH_FLOOD
+                ) {
+                    return {
+                        is_man_made_event: {},
+                    };
+                }
+                return {
+                    is_man_made_event: { forceValue: nullValue },
+                };
+            },
+        );
+
+        formFields = addCondition(
+            formFields,
+            formValue,
+            ['type_of_dref'],
+            [
+                'people_in_need',
+                'amount_requested',
+                'emergency_appeal_planned',
+                'event_map_file',
+                'cover_image_file',
+            ],
+            (val) => {
```

Comment:
Shouldn't we define this as const?
Or is that inferred in the library?

Reply:
We do not need to use "as const" as it's inferred from the library.
But, we should still define the return type of the last function in addCondition as typescript does not support exact types